main.tf

```diff
@@ -26,7 +36,8 @@ resource "aws_vpc" "this" {
   enable_dns_support               = var.enable_dns_support
   enable_classiclink               = var.enable_classiclink
   enable_classiclink_dns_support   = var.enable_classiclink_dns_support
-  assign_generated_ipv6_cidr_block = var.enable_ipv6
+  assign_generated_ipv6_cidr_block = local.assign_generated_ipv6_cidr_block
```

Comment:
If I do this, then I run into `Error: Conflicting configuration arguments` when I set `ipv6_ipam_pool_id` because either `assign_generated_ipv6_cidr_block` needs to be `null` or `ipv6_ipam_pool_id` needs to be `null`.  Having one as `true` and the other as `false` isn't good enough unfortunately.

See

```NOFORMAT
hnicholas@hnicholas-a01:~/.local/tmp/VCTO-4282-Reconsider-Terragrunt/backend-nonprod/us-west-2/backend-nonprod/vpc$ terragrunt plan
[INFO] Getting version from tgenv-version-name
[INFO] TGENV_VERSION is 0.37.1
module.vpc.aws_vpc.this[0]: Refreshing state... [id=vpc-0aa9cd4bdee64110c]
aws_eip.nat[1]: Refreshing state... [id=eipalloc-09409fd984a3a84de]
aws_eip.nat[3]: Refreshing state... [id=eipalloc-08ef20e5a8c8ff5a1]
aws_eip.nat[2]: Refreshing state... [id=eipalloc-009092ece843788a5]
aws_eip.nat[0]: Refreshing state... [id=eipalloc-0f152827258454fe0]

Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 29, in resource "aws_vpc" "this":
  29:   assign_generated_ipv6_cidr_block = var.enable_ipv6 && var.assign_generated_ipv6_cidr_block

"assign_generated_ipv6_cidr_block": conflicts with ipv6_ipam_pool_id


Error: Conflicting configuration arguments

  on .terraform/modules/vpc/main.tf line 30, in resource "aws_vpc" "this":
  30:   ipv6_ipam_pool_id                = var.ipv6_ipam_pool_id

"ipv6_ipam_pool_id": conflicts with assign_generated_ipv6_cidr_block

Releasing state lock. This may take a few moments...
ERRO[0008] 1 error occurred:
        * exit status 1
```

Reply:
How about this change?

```suggestion
  assign_generated_ipv6_cidr_block = var.enable_ipv6 && var.assign_generated_ipv6_cidr_block ? true : null
```

and set `default` value of `assign_generated_ipv6_cidr_block = false`.

I have just verified it by running `example/simple-vpc`, and it seems to work as expected. There are no backward-compatibility breaks.

Reply:
Not sure why, but when I do that, then it wants to have AWS generate the IPv6 CIDR block rather than using the `ipv6_ipam_pool_id` that I pass in 🤷 

```NOFORMAT
  # module.vpc.aws_vpc.this[0] will be updated in-place
  ~ resource "aws_vpc" "this" {
      - assign_generated_ipv6_cidr_block     = true -> null
        id                                   = "vpc-0ece25ad5cc48c0bc"
      ~ ipv6_association_id                  = "vpc-cidr-assoc-0b767bf40a41cfabe" -> (known after apply)
      ~ ipv6_cidr_block                      = "2406:da18:97:cb00::/56" -> (known after apply)
        tags                                 = {
            "Contact"       = "dmVsby1vcHNAdm13YXJlLmNvbTsgI3ZlbG8tbm9jLWFzc2lzdA=="
            "Customer"      = "Backend"
            "Environment"   = "backend-nonprod"
            "Name"          = "backend-nonprod-ap-southeast-1-vpc"
            "ProvisionedBy" = "https://gitlab.eng.vmware.com/velo-tech-ops/terraform/modules/aws/vpc"
            "Region"        = "aws-ap-southeast-1"
        }
        # (16 unchanged attributes hidden)
    }
```

Reply:
Done!